addons/hr_work_entry_contract/models/hr_contract.py

```diff
@@ -82,8 +82,8 @@ def _get_sub_leave_domain(self):
     def _get_leave_domain(self, start_dt, end_dt):
         domain = [
             ('resource_id', 'in', [False] + self.employee_id.resource_id.ids),
-            ('date_from', '<=', end_dt),
-            ('date_to', '>=', start_dt),
+            ('date_from', '<=', end_dt.replace(tzinfo=None)),
+            ('date_to', '>=', start_dt.replace(tzinfo=None)),
```

Comment:
Shouldn't this be managed by the framework? It feels like needing to set a date back to UTC every time from the domain seems error prone :thinking:

Reply:
Here, I am making the datetime naive (without a timezone). All framework expects naive datetimes and you already have to convert the datetime to UTC then to naive (for example to assign it to a field); actually only the value in domains was accepting localized datetimes by accident.
Timezones are generally error-prone and having to handle them explicitly makes the code more robust. We will re-discuss in the team whether we impose this or not.

Reply:
Ok, thanks for the info